src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,32 @@ pub trait Iterator {
         None
     }
 
+    /// Returns an iterator adaptor that goes over the original iterator,
+    /// skipping `step` elements at a time.
+    ///
+    /// **Panics** if `step == 0`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let a = [0, 1, 2, 3, 4, 5];
+    ///
+    /// let mut iter = a.iter().every_nth(2);
+    ///
+    /// assert_eq!(iter.next(), Some(&0));
+    /// assert_eq!(iter.next(), Some(&2));
+    /// assert_eq!(iter.next(), Some(&4));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
Does taking the name `step_by` really require _removing_ the `Step` trait?

Removing `step`, `steps_between`, and `is_negative` from `Step` seems logical, as is replacing `core::iter::StepBy` with the logic worked out here.  But redoing all the `Range` iterators feels like it shouldn't be necessary for this change, and I don't see how the existence of `Step` conflicts with it.